src/Application.php

```diff
@@ -639,33 +468,15 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
-            ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
+        if ($this->container && $this->container->has(Middleware\ErrorResponseGenerator::class)) {
```

Comment:
OK. I see where you are coming from. I think focusing on the listener logic made me overlook the fact that `ErrorHandler` is in fact just middleware and should not be treated like a service. I agree that this is a very edge case that is much like an `Exception` being thrown because the container has an error, missing config, etc. I can modify my already custom `ErrorResponseGenerator` to replicate the `ErrorHandler` listener logic and attach to there instead of onto the middleware.

Perhaps at a later date it can be discussed if the listener logic should be moved to `ErrorResponseGenerator` in Stratigility to help bring `ErrorHandler` to a more "pure" middleware class without ancillary methods.